gpage.h

```diff
@@ -222,16 +228,17 @@ namespace gcpp {
 	//
 	inline
 	bool gpage::contains(gsl::not_null<const byte*> p) const noexcept {
-		return &storage[0] <= p && p < &storage[total_size - 1];
+		auto cmp = std::less<>{};
+		return !cmp(p, &storage[0]) && cmp(p, &storage[total_size - 1]);
```

Comment:
We ought to include the final byte even though we don't allocate it, for the same reason we don't allocate it -- one-past-the-end pointers should work.